### DISCUSSION TITLE
How does conversation-topic-splitter works?

### DISCUSSION THREAD

**Discussion by OnlinePage** [aurelio-labs/semantic-router#218] [Q&A]
I am curious about working of conversation-topic-splitter ?

Like if we only sent the user current message to filter route then its not good way , cuz most of the time it will just predict wrong router as it does not have context for last messages between user and the bot.

https://github.com/aurelio-labs/semantic-router/blob/main/docs/examples/conversation-topic-splitter.ipynb

Now since i came across 👆, I m curious how's does this helps in routing with context ? or is there any better approach?🤔

Is this the good way to have proper and reliable routing in an agent?

@jamescalam

**Comment by octocat20** [comment]
Yeah routing on only the last user message is brittle. You lose the earlier turns that actually define the topic.

That notebook is basically a preprocessing trick. It splits the chat into topic chunks by embedding similarity, then you feed the *current topic* into the router instead of one isolated line.

So "Yeah, v.3.11.2" alone looks like a version question. With the earlier "computer keeps crashing" turns in the same topic, the router has something usable.

It is not a special routing mode. Pattern is:
1. keep history
2. split into topics as the chat moves
3. route on the current topic text

That is a good approach for agents. A sliding window of recent turns or an LLM rewrite of the latest message with resolved context works too.

Side note: the old conversation-topic-splitter.ipynb path may be gone on current main. The idea still holds if you build the topic window yourself.
